File: exemplos/aula_250617/example3.java
import java.util.Scanner;

class SenhaInvalidaError extends Exception{
    public SenhaInvalidaError() {
        super("Senha insegura");
    }

}

class Login{
    public static void validarSenha(String senha) throws SenhaInvalidaError {
        if (senha.length() < 8 ) {
            throw new SenhaInvalidaError();
        }
    }
}

class Example3 {
    public static void main(String[] args) {
        System.out.print("Digite uma senha: ");
        Scanner scanner = new Scanner(System.in);
        String senha = scanner.nextLine();
        scanner.close();
        try {
            Login.validarSenha(senha);
            System.out.println("Senha válida");
        } catch (SenhaInvalidaError e) {
            System.out.println(e.getMessage());
        }
    }
}

File: exemplos/aula_250617/example3.py
class SenhaInvalidaError(Exception):
    """Exceção para senhas inválidas."""
    def __init__(self):
        super().__init__("Senha insegura")
        
class Login:
    @staticmethod
    def validar_senha(senha):
        """Valida a senha de acordo com os critérios especificados."""
        if len(senha) < 8:
            raise SenhaInvalidaError()


try:
    Login.validar_senha(input("Digite uma senha: "))
    print("Senha válida! ✅")
except SenhaInvalidaError as e:
    print(f"Erro: {e} 🚫")